packages/core/src/react-integration/__tests__/integration-optimistic-endpoint.web.tsx

```diff
@@ -406,8 +409,167 @@ describe.each([
       expect(result.current.article).toEqual(
         CoolerArticleResource.fromJS({
           ...payload,
-          title: 'first',
+          title: 'second',
+          tags: ['thirdoptimistic'],
+        }),
+      );
+    });
+
+    it('should clear optimistic when server response resolves in order', async () => {
```

Comment:
@sweetro This test case covers a server responding to each request in order. As you can see with the expects - each time we stop applying the next optimistic response